code/game/objects/random/random.dm

```diff
@@ -1123,4 +1123,18 @@ var/list/multi_point_spawns
 				/obj/item/clothing/suit/space/void/mining/alt,
 				/obj/item/clothing/head/helmet/space/void/mining/alt
 			)
-		)
\ No newline at end of file
+		)
+
+/obj/random/landmine
+	name = "Random Landmine"
+	desc = "This is a random landmine."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "uglymine"
+	spawn_nothing_percentage = 25
+
+/obj/random/landmine/item_to_spawn()
+	return pick(prob(30);/obj/effect/mine,
+				prob(25);/obj/effect/mine/frag,
+				prob(25);/obj/effect/mine/emp,
+				prob(10);/obj/effect/mine/stun,
```

Comment:
Last I checked, stun mines will still freeze you on the spot rather than knocking you down? Might be worth looking into.

Reply:
In an idea world it'd literally be a flash bang you can step on, but I couldn't get the grenade-mines to go off without literally deleting every external organ on the mob.